src/main/java/nextstep/subway/domain/Sections.java

```diff
@@ -0,0 +1,125 @@
+package nextstep.subway.domain;
+
+import nextstep.subway.domain.exceptions.CanNotAddSectionException;
+
+import javax.persistence.CascadeType;
+import javax.persistence.Embeddable;
+import javax.persistence.OneToMany;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Objects;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+@Embeddable
+public class Sections {
+
+    @OneToMany(mappedBy = "line", cascade = {CascadeType.PERSIST, CascadeType.MERGE}, orphanRemoval = true)
+    private List<Section> sections = new ArrayList<>();
+
+    public Sections() {
+
+    }
+
+    public void addSection(Section newSection) {
+        if (sections.isEmpty()) {
+            sections.add(newSection);
+            return;
+        }
+
+        Station newUpStation = newSection.getUpStation();
+        Station newDownStation = newSection.getDownStation();
+
+        if (hasStation(newUpStation) && hasStation(newDownStation)) {
+            throw new CanNotAddSectionException("상행역과 하행역 모두 이미 존재하여 구간 추가 불가능");
+        }
+
+        if (!hasStation(newUpStation) && !hasStation(newDownStation)) {
+            throw new CanNotAddSectionException("상행역과 하행역 모두 존재하지 않아 구간 추가 불가능");
+        }
+
+        if (isFirstStation(newDownStation)) {
+            sections.add(newSection);
+            return;
+        }
+
+        if (isLastStation(newUpStation)) {
+            sections.add(newSection);
+            return;
+        }
+
+        Section ordinarySection = sections.stream()
+                .filter(it -> it.getUpStation().equals(newUpStation) || it.getDownStation().equals(newDownStation))
+                .findFirst().get();
```

Comment:
```suggestion
                .findFirst().orElseThrow(exception);
```

get 보다는 orElseThrow 와 같은 메서드들을 사용하시는것을 추천드립니다 👍 

[참고](http://homoefficio.github.io/2019/10/03/Java-Optional-%EB%B0%94%EB%A5%B4%EA%B2%8C-%EC%93%B0%EA%B8%B0/)

Reply:
저도 동의해요.

`get()` 을 사용할 때 당장엔 예외가 발생하진 않겠지만, 

추후 유지보수 과정에서 실수로 sections의 불변식이 깨지는 경우 

언제든지 예외가 발생할 수 있기에 개인적으로는 빠른 디버깅을 위해 예외를 고려하는 편이긴 합니다.

좋은 코멘트 감사합니다. 😊 [d3144ae](https://github.com/next-step/atdd-subway-path/pull/486/commits/d3144aea2616ba97dc011395de95d062719d4f49)